src/ECS/Components/Transform.h

```diff
@@ -22,4 +24,6 @@ struct Transform
 	glm::vec3 scale;
 };
 
+const auto k_GetTransformPosition = [](Transform hit) constexpr -> std::optional<glm::vec3> { return hit.position; };
```

Comment:
At this point it doesn't have to be a lambda and can be an inline function. Also make sure hit is a reference to avoid needlessly copying it.